sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ParDoTest.java

```diff
@@ -2018,6 +2018,34 @@ public void onTimer(OnTimerContext context) {
 
   @Test
   @Category({ValidatesRunner.class, UsesTimersInParDo.class})
+  public void testEventTimeTimerAlignBounded() throws Exception {
+    final String timerId = "foo";
+
+    DoFn<KV<String, Integer>, Integer> fn =
+        new DoFn<KV<String, Integer>, Integer>() {
+
+          @TimerId(timerId)
+          private final TimerSpec spec = TimerSpecs.timer(TimeDomain.EVENT_TIME);
+
+          @ProcessElement
+          public void processElement(ProcessContext context, @TimerId(timerId) Timer timer) {
+            timer.setForNowAlign(Duration.standardSeconds(1), Duration.millis(1));
```

Comment:
How about this: 
```java
  void set(Instant instant);
  void setForNow();
  Timer plus(Duration duration);
  Timer align(Duration duration);
  Timer offset(Duration duration);
```

User can do these:  
`timer.alignTo(duration).offset(duration).set(targetInstant)`  
`timer.alignTo(duration).offset(duration).setForNow()`  
`timer.plus(duration).set(targetInstant)`  
`timer.plus(duration).setForNow()`  
`timer.set(targetInstant)`  
`timer.setForNow()`